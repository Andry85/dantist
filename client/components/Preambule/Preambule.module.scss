@import './../../styles/variables';

.preambule {
    display: grid;
    grid-template-columns: 50% 1fr;
    gap: 1.5rem;

    @include breakpoint(md) {
        grid-template-columns: 1fr;
    }

    &__text {
        padding: 4rem 2rem;

        p {
            margin-bottom: 1.6rem;
            font-size: 1.8rem;
            line-height: 1.5;
        }
    }

    &__more {
        color: $base_color;
        border: 2px solid $base_color;
        padding: 1rem 2rem;
        display: inline-block;
        text-decoration: none;
        margin: 2rem 0 0 0;

        &:hover {
            border-color: $dark-color;
            color: $dark-color;
        }
    }
}