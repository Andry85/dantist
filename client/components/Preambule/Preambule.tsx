import styles from './Preambule.module.scss';
import Image from 'next/image';
import Link from 'next/link';



type PreambuleProps = {
    preambule: {
        image: string;
        description: string;
        button: {
         title: string;
         link: string;
        }
    }
  
}

const Preambule = ({preambule} : PreambuleProps) => {
   


    return (
        <>
            <article className={styles.preambule}>
                <div className={styles.preambule__pic}>
                    <Image src={preambule.image} width={1280} height={848} alt="" />
                </div>
                <div className={styles.preambule__text}>
                    <p>{preambule.description}</p>
                    <Link className={styles.preambule__more}  href={preambule.button.link}>{preambule.button.title}</Link>
                </div>
            </article>
        
        </>
    )
}


export default Preambule;