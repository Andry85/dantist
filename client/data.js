export const menu = [
    {
        id: 0,
        title: "Main",
        link: "/"
    },
    {
        id: 1,
        title: "About Doctor",
        link: "/about"
    },
    {
        id: 2,
        title: "FAQ",
        link: "/faq"
    }
]

export const contacts = {
  number: "(555) 555-1234",
  email: "smile@dantist.com"
}

export const preambule = {
    image: "/images/dentist-about.jpg",
    description: "Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard du Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy mmy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book. It has survived not only five centuries",
    button: {
        title: "Mor details...",
        link: "/about"
    }
}

