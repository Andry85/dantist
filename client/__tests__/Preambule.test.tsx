import { render, screen } from '@testing-library/react'
import renderer from 'react-test-renderer';
import Preambule from '../components/Preambule/Preambule';
import '@testing-library/jest-dom'
import {preambule} from '../data'



describe('Preambule', () => {
  it('renders text desc', () => {
    render( <Preambule preambule={preambule} />)
    const preambuletext = screen.getByText(/Lorem Ipsum/i);
    expect(preambuletext).toBeInTheDocument()
  })
})

it('renders correctly', () => {
  const tree = renderer
    .create(<Preambule preambule={preambule}/>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});