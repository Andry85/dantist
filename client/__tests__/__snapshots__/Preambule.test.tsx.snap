// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<article
  className="preambule"
>
  <div
    className="preambule__pic"
  >
    <img
      alt=""
      data-nimg="1"
      decoding="async"
      height={848}
      loading="lazy"
      onError={[Function]}
      onLoad={[Function]}
      src="/_next/image?url=%2Fimages%2Fdentist-about.jpg&w=3840&q=75"
      srcSet="/_next/image?url=%2Fimages%2Fdentist-about.jpg&w=1920&q=75 1x, /_next/image?url=%2Fimages%2Fdentist-about.jpg&w=3840&q=75 2x"
      style={
        {
          "color": "transparent",
        }
      }
      width={1280}
    />
  </div>
  <div
    className="preambule__text"
  >
    <p>
      Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard du Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy mmy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book. It has survived not only five centuries
    </p>
    <a
      className="preambule__more"
      href="/about"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onTouchStart={[Function]}
    >
      Mor details...
    </a>
  </div>
</article>
`;
